tests/test_options.py

```diff
@@ -212,3 +264,365 @@ def test_as_dict():
+
+
+def test_initialize_options_source_dict_2():
+    reset_options_instance()
+    sys.argv += ['-o', 'tests/default.yaml', '--model.network', 'mynet']
+    source = {
+        'dataset': 123,
+        'model': {
+            'criterion': 'I am a criterion',
+            'network': 'I am a network',
+        },
+    }
+    Options(source, run_parser=True)
+    assert Options()['model']['network'] == 'mynet'
+
+
+def test_initialize_options_source_dict_3():
+    reset_options_instance()
+    source1 = {
+        'dataset': 123,
+        'model': {
+            'criterion': 'I am a criterion',
+            'network': 'I am a network',
+        },
+    }
+    Options(source1, run_parser=False)
+    assert Options().options == OptionsDict(source1)
+    assert Options().source == source1
+
+    source2 = {
+        'Micael': 'is the best',
+        'Remi': 'is awesome',
+    }
+    Options(source2, run_parser=False)
+    assert Options().options == OptionsDict(source1)
+    assert Options().source == source1
+
+
+def test_initialize_options_source_dict_4():
+    reset_options_instance()
+    source = {
+        'dataset': 123,
+        'model': {
+            'criterion': 'I am a criterion',
+            'network': 'I am a network',
+        },
+    }
+    with pytest.raises(SystemExit):
+        Options(source, run_parser=True)
+
+
+def test_initialize_options_source_optionsdict():
+    reset_options_instance()
+    source = OptionsDict({
+        'dataset': 124,
+        'model': {
+            'criterion': 'I am a criterion',
+            'network': 'I am a network',
+        },
+    })
+    Options(source, run_parser=False)
+    assert Options().options == source
+    assert Options().source == source.asdict()
+
+
+def test_initialize_options_incorrect_source():
+    reset_options_instance()
+    source = 123
+    with pytest.raises(TypeError):
+        Options(source, run_parser=False)
+
+
+def test_initialize_arguments_callback():
+    reset_options_instance()
+    sys.argv += ['-o', 'tests/default.yaml']
+    source = {
+        'dataset': 'mydataset',
+        'model': 'mymodel',
+    }
+
+    def arguments_callback_a(instance, arguments, options_dict):
+        arguments.dataset = arguments.dataset + 'a'
+        arguments.model = arguments.model + 'a'
+        return arguments
+
+    Options(source, arguments_callback=arguments_callback_a)
+    source_a = {
+        'path_opts': 'tests/default.yaml',
+        'dataset': 'mydataseta',
+        'model': 'mymodela',
+    }
+    assert Options().options == OptionsDict(source_a)
+    assert Options().source == source
+
+
+def test_initialize_lock():
+    reset_options_instance()
+    source = {
+        'dataset': 123,
+        'model': {
+            'criterion': 'I am a criterion',
+            'network': 'I am a network',
+        },
+    }
+    Options(source, run_parser=False, lock=True)
+    assert Options().options.islocked()
+
+
+def test_initialize_unlock():
```

Comment:
Let's rename this to `test_initialize_not_locked`?
Because there is actually a function called `unlock` and it is supposed to print a lot of stuff to inform the user Options has been unlocked. It is a safety mechanism